Problems while implementing #3963

Had to change sorted_entries implementation because of this issue.

```
./build/lib/container/ordered_map.fz:76:27: error 1: Incompatible types when passing argument in a call
      .find_by_comparator (e -> ks[e.i] ⋄ k)

Actual type for argument #1 'cmp' does not match expected type.
In call to          : 'container.sorted_array.find_by_comparator'
expected formal type: 'Unary i32 (container.ordered_map.this container.ordered_map.OK container.ordered_map.V).entry'
actual type found   : '((container.this.ordered_map.this container.ordered_map.OK container.ordered_map.V).index [ ].this).λ'
assignable to       : '((container.this.ordered_map.this container.ordered_map.OK container.ordered_map.V).index [ ].this).λ',
                      'Any',
                      'Function i32 (container.this.ordered_map.this container.ordered_map.OK container.ordered_map.V).entry',
                      'Unary i32 (container.this.ordered_map.this container.ordered_map.OK container.ordered_map.V).entry'
for value assigned  : '(e -> ks[e.i] ⋄ k)'
To solve this, you could change the type of the target 'cmp' to '((container.this.ordered_map.this container.ordered_map.OK container.ordered_map.V).index [ ].this).λ' or convert the type of the assigned value to 'Unary i32 (container.ordered_map.this container.ordered_map.OK container.ordered_map.V).entry'.


./build/lib/container/ordered_map.fz:82:66: error 2: Incompatible types when passing argument in a call
  public redef items Sequence (tuple OK V) => sorted_entries.map e->(e.key, e.val)

Actual type for argument #1 'f' does not match expected type.
In call to          : 'Sequence.map'
expected formal type: 'Unary (tuple container.ordered_map.OK container.ordered_map.V) (container.ordered_map.this container.ordered_map.OK container.ordered_map.V).entry'
actual type found   : '(container.this.ordered_map.this container.ordered_map.OK container.ordered_map.V).items.this.λ'
assignable to       : '(container.this.ordered_map.this container.ordered_map.OK container.ordered_map.V).items.this.λ',
                      'Any',
                      'Function (tuple container.ordered_map.OK container.ordered_map.V) (container.this.ordered_map.this container.ordered_map.OK container.ordered_map.V).entry',
                      'Unary (tuple container.ordered_map.OK container.ordered_map.V) (container.this.ordered_map.this container.ordered_map.OK container.ordered_map.V).entry'
for value assigned  : 'e->(e.key, e.val)'
To solve this, you could change the type of the target 'f' to '(container.this.ordered_map.this container.ordered_map.OK container.ordered_map.V).items.this.λ' or convert the type of the assigned value to 'Unary (tuple container.ordered_map.OK container.ordered_map.V) (container.ordered_map.this container.ordered_map.OK container.ordered_map.V).entry'.

2 errors.
make: *** [Makefile:619: build/modules/base.fum] Error 1
```

--- a/modules/base/src/container/ordered_map.fz
+++ b/modules/base/src/container/ordered_map.fz
@@ -63,35 +63,29 @@ is
 
   # a sorted array of entries of this map
   #
-  public sorted_entries =>
-    sorted_indices
-      .map (i -> entry i)
-      .as_array
-
-
-  # a sorted array of indices of this map
-  #
-  sorted_indices := ks.indices.sort_by (a,b)->ks[a]≤ks[b]
+  public sorted_entries :=
+    ks.indices.map (i -> entry i)
+      .sort
 
 
   # number of entries in this map
   #
-  public redef size i32 => sorted_indices.length
+  public redef size i32 => sorted_entries.length
 
 
   # get the value k is mapped to, or nil if none.
   #
   # performance is O(log size).
   #
-  public redef index [] (k OK) option V=>
-    sorted_indices
-      .find_by_comparator (i -> ks[i] ⋄ k)
+  public redef index [] (k OK) option V =>
+    sorted_entries
+      .find_by_comparator (e -> ks[e.i] ⋄ k)
       .bind i->sorted_entries[i].val
 
 
   # get an array of all key/value pairs in this map
   #
-  public redef items Sequence (tuple OK V) => sorted_indices.map i->(ks[i], vs[i])
+  public redef items Sequence (tuple OK V) => sorted_entries.map e->(e.key, e.val)
 
 
   # add mapping from k to v

